Storage/Storage/CoreData/CoreDataIterativeMigrator.swift

```diff
@@ -102,7 +102,16 @@ public struct CoreDataIterativeMigrator {
             debugMessages.append(migrationAttemptMessage)
             DDLogWarn(migrationAttemptMessage)
 
-            guard migrateStore(at: sourceStore, storeType: storeType, fromModel: modelFrom, toModel: modelTo, with: migrateWithModel) == true else {
+            let (success, migrateStoreError) = migrateStore(at: sourceStore,
```

Comment:
Not related to this line. 

I noticed that the errors we were getting are missing reasons for the failed migrations. For example:

![image](https://user-images.githubusercontent.com/198826/86023590-1c356000-b9e9-11ea-83c9-448fdce97157.png)

It looks like we don't return any debugging messages for cases where we return `false`. Example: 

https://github.com/woocommerce/woocommerce-ios/blob/4350b456efd71fd1f3956a58fda997c0b3573d8a/Storage/Storage/CoreData/CoreDataIterativeMigrator.swift#L31-L33

Do you think we should add messages for those?

Reply:
Looking more closely, it looks like this will be handled by the `do catch`.

Reply:
there are many points of failure in this function `CoreDataIterativeMigrator. iterativeMigrate`, and I think the most critical one is probably where it migrates the store in this call to `migrateStore`. later in this file I updated to return the error for each point of failure (actual migration error, or backup error) and adding this error to `debugMessages` below (L113)